test/rpc-test.js

```diff
@@ -0,0 +1,227 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const consensus = require('../lib/protocol/consensus');
+const FullNode = require('../lib/node/fullnode');
+const Network = require('../lib/protocol/network');
+const network = Network.get('regtest');
+
+const node = new FullNode({
+  network: 'regtest',
+  apiKey: 'foo',
+  walletAuth: true,
+  memory: true,
+  workers: true,
+  plugins: [require('../lib/wallet/plugin')]
+});
+
+const {NodeClient, WalletClient} = require('bclient');
+
+const nclient = new NodeClient({
+  port: network.rpcPort,
+  apiKey: 'foo'
+});
+
+const wclient = new WalletClient({
+  port: network.walletPort,
+  apiKey: 'foo'
+});
+
+const {wdb} = node.require('walletdb');
+
+let addressHot = null;
+let addressMiner = null;
+let walletHot = null;
+let walletMiner = null;
+let blocks = null;
+let txid = null;
+let utxo = null;
+
+describe('RPC', function() {
+  this.timeout(15000);
+
+  it('should open node and create wallets', async () => {
+    consensus.COINBASE_MATURITY = 0;
+    await node.open();
+    await nclient.open();
+    await wclient.open();
+
+    const walletHotInfo = await wclient.createWallet('hot');
+    walletHot = wclient.wallet('hot', walletHotInfo.token);
+    const walletMinerInfo = await wclient.createWallet('miner');
+    walletMiner = wclient.wallet('miner', walletMinerInfo.token);
+    await walletHot.open();
+    await walletMiner.open();
+  });
+
+  it('should rpc help', async () => {
+    assert( await nclient.execute('help', []) );
+    assert( await wclient.execute('help', []) );
+
+    await assert.asyncThrows( async () => await nclient.execute('help',
+      ['getinfo']), 'getinfo');
+
+    await assert.asyncThrows( async () => await wclient.execute('help',
+      ['getbalance']), 'getbalance');
+  });
+
+  it('should rpc getinfo', async () => {
+    const info = await nclient.execute('getinfo', []);
+    assert.strictEqual(info.blocks, 0);
+  });
+
+  it('should rpc selectwallet', async () => {
+    const response = await wclient.execute('selectwallet', ['miner']);
+    assert.strictEqual(response, null);
+  });
+
+  it('should rpc getnewaddress from default account', async () => {
+    const acctAddr = await wclient.execute('getnewaddress', []);
+    assert(acctAddr);
+  });
+
+  it('should fail rpc getnewaddress from nonexistent account', async () => {
+    await assert.asyncThrows( async () =>
+      await wclient.execute('getnewaddress', ['bad-account-name']),
+      'Account not found.');
+  });
+
+  it('should rpc getaccountaddress', async () => {
+    addressMiner = await wclient.execute('getaccountaddress', ['default']);
+    assert(addressMiner);
+  });
+
+  it('should rpc generatetoaddress', async () => {
+    blocks = await nclient.execute('generatetoaddress',
```

Comment:
If the previous test fails this test will fail in a cascade... may be a way to have it be independent?

Reply:
Well, the test needs some funds to spend... Does it make more sense to create a new wallet, address, and coinbase tx for each subsequent test?

Reply:
Not usually necessary for each test, sub `descriibes` can all share a similar required setup via a `before` block. And the `describe` block can fail or be skipped without affecting the other tests, as similar each of the `it` blocks within it.

Reply:
We'll have to revisit this later, and look at updating some of the other tests too.